packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1346,12 +1346,18 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
   /// the platform with [NavigationNotification.canHandlePop] and stops
   /// bubbling.
   bool _defaultOnNavigationNotification(NavigationNotification notification) {
-    // Don't do anything with navigation notifications if there is no engine
-    // attached.
-    if (_appLifecycleState != AppLifecycleState.detached) {
-      SystemNavigator.setFrameworkHandlesBack(notification.canHandlePop);
+    switch(_appLifecycleState) {
+      case null:
+      case AppLifecycleState.detached:
+      case AppLifecycleState.inactive:
```

Comment:
I then observed in the failing tests that the problematic calls were only made when the lifecycle was `inactive`.

Reply:
should this be handle on the engine side? it seems hard for the caller of `SystemNavigator.setFrameworkHandlesBack` to know this.

Reply:
In my experimentation, the failure happens even if the engine doesn't handle the call at all.  I've been super confused about how those tests are set up that this kind of timeout error is even possible.  I would definitely be up for revisiting this code if someone finds a better way to avoid the failure.